src/Formula.elm

```diff
@@ -223,9 +362,29 @@ mapResult f =
     List.foldr (Result.map2 (::) << f) (Ok [])
 
 
+removeQuantifierAndSubstitute : Substitution -> Formula -> Result String Formula
+removeQuantifierAndSubstitute substitution original =
+    case original of
+        ForAll s f ->
+            if List.member s (Dict.keys substitution) then
+                removeQuantifierAndSubstitute substitution f
+            else
+                substFormula substitution f |> Result.map (ForAll s)
```

Comment:
Tu to teda asi chce chybu ...
(A plust teda mozno celkovo aj ked `substitution` ma viac premennych)